src/fairseq2/sharder.py

```diff
@@ -29,7 +32,20 @@
 @dataclass
 class ShardSpec:
     dim: int
+    """The sharded dimension."""
+
     region_boundary: bool = False
+    """Whether the sharded dimension is at the boundary between
+    two model parallel regions. Allows to avoid unnecessary communication."""
+
+    disable_end_reduce: bool = False
+    """Whether to disable a reduce planned at the end
+    of the forward of a sharded layer.
+    If ``True``, it is the responsibility of the layer's user to perform the reduce."""
+
+    shard_children: bool = False
```

Comment:
We likely won't need this. Let me think more about the sharding APIs. In the meantime it can stay here as it is.